numpy/ma/core.py

```diff
@@ -2294,11 +2294,15 @@ def _recursive_printoption(result, mask, printopt):
     """
     names = result.dtype.names
     for name in names:
-        (curdata, curmask) = (result[name], mask[name])
-        if curdata.dtype.names:
-            _recursive_printoption(curdata, curmask, printopt)
-        else:
-            np.copyto(curdata, printopt, where=curmask)
+        curdata, curmask = result[name], mask[name]
+        if isinstance(curdata, ndarray):
+            if curdata.dtype.names:
+                _recursive_printoption(curdata, curmask, printopt)
+            else:
+                np.copyto(curdata, printopt, where=curmask)
+        else:  # scalar values
```

Comment:
This fails for nested structs, eg

    >>> a = np.ma.array([(1,(1,1))], mask=[(True,(True,True))], dtype = [('a', '<i8'), ('b', 'i4,i4')])
    >>> str(a)
    ValueError: Setting void-array with object members using buffer.

I'm also a little worried that the first if-block will catch too many things it shouldn't. It's not immediately clear what the code should be.